diff --git a/socialNetwork/src/UniqueIdService/UniqueIdHandler.h b/socialNetwork/src/UniqueIdService/UniqueIdHandler.h
index 9c59c69..f81aaa1 100644
--- a/socialNetwork/src/UniqueIdService/UniqueIdHandler.h
+++ b/socialNetwork/src/UniqueIdService/UniqueIdHandler.h
@@ -20,14 +20,16 @@ namespace social_network {
 
 using std::chrono::duration_cast;
 using std::chrono::milliseconds;
-using std::chrono::system_clock;
+using std::chrono::steady_clock;
 
 static int64_t current_timestamp = -1;
 static int counter = 0;
 
 static int GetCounter(int64_t timestamp) {
   if (current_timestamp > timestamp) {
-    LOG(fatal) << "Timestamps are not incremental.";
+    LOG(fatal) << "Timestamps are not incremental." <<
+        " Current timestamp: " << current_timestamp <<
+        ", next timestamp: " << timestamp;
     exit(EXIT_FAILURE);
   }
   if (current_timestamp == timestamp) {
@@ -72,9 +74,8 @@ int64_t UniqueIdHandler::ComposeUniqueId(
 
   _thread_lock->lock();
   int64_t timestamp =
-      duration_cast<milliseconds>(system_clock::now().time_since_epoch())
-          .count() -
-      CUSTOM_EPOCH;
+      duration_cast<milliseconds>(steady_clock::now().time_since_epoch())
+          .count();
   int idx = GetCounter(timestamp);
   _thread_lock->unlock();
 
diff --git a/socialNetwork/src/UserService/UserHandler.h b/socialNetwork/src/UserService/UserHandler.h
index ced466b..b91bf93 100644
--- a/socialNetwork/src/UserService/UserHandler.h
+++ b/socialNetwork/src/UserService/UserHandler.h
@@ -32,6 +32,7 @@ using std::chrono::duration_cast;
 using std::chrono::milliseconds;
 using std::chrono::seconds;
 using std::chrono::system_clock;
+using std::chrono::steady_clock;
 using namespace jwt::params;
 
 static int64_t current_timestamp = -1;
@@ -39,7 +40,9 @@ static int counter = 0;
 
 static int GetCounter(int64_t timestamp) {
   if (current_timestamp > timestamp) {
-    LOG(fatal) << "Timestamps are not incremental.";
+    LOG(fatal) << "Timestamps are not incremental." <<
+        " Current timestamp: " << current_timestamp <<
+        ", next timestamp: " << timestamp;
     exit(EXIT_FAILURE);
   }
   if (current_timestamp == timestamp) {
@@ -246,9 +249,8 @@ void UserHandler::RegisterUser(
   // Compose user_id
   _thread_lock->lock();
   int64_t timestamp =
-      duration_cast<milliseconds>(system_clock::now().time_since_epoch())
-          .count() -
-      CUSTOM_EPOCH;
+      duration_cast<milliseconds>(steady_clock::now().time_since_epoch())
+          .count();
   int idx = GetCounter(timestamp);
   _thread_lock->unlock();
 
